src/main/groovy/pl/allegro/tech/build/axion/release/ReleaseTask.groovy

```diff
@@ -15,10 +16,14 @@ abstract class ReleaseTask extends BaseAxionTask {
     void release() {
         VersionResolutionContext context = resolutionContext()
         Releaser releaser = context.releaser()
-        ScmPushResult result = releaser.releaseAndPush(context.rules())
+        ScmService scmService = context.scmService()
+        def releaseBranchNames = scmService.getReleaseBranchNames()
+        def currentBranch = context.repository().currentPosition().getBranch()
+        def isReleaseOnlyOnDefaultBranches = scmService.isReleaseOnlyOnDefaultBranches()
+        ScmPushResult result = releaser.releaseAndPush(context.rules(), isReleaseOnlyOnDefaultBranches, currentBranch, releaseBranchNames)
```

Comment:
I would pass only `context` to `relaseAndPush` and extract everything inside `Releaser`. Methods with many parameters are hard to reason about and also now we have duplication in `ReleaseTask` and `CreateReleaseTask`.

Reply:
we've agreed on object argument ;)